modules/nlu/src/backend/application/bot/index.ts

```diff
@@ -0,0 +1,78 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import { DefinitionsRepository } from '../definitions-repository'
+import { BotDoesntSpeakLanguageError } from '../errors'
+import { ModelEntryService, TrainingEntryService } from '../model-entry'
+import { NLUClient } from '../nlu-client'
+import { TrainingState, TrainingSession, BotDefinition } from '../typings'
+import { Predictor } from './predictor'
+import { Trainer } from './trainer'
+
+export interface MountOptions {
+  queueTraining: boolean
+}
+
+export class Bot {
+  private _botId: string
+  private _languages: string[]
+
+  private _trainer: Trainer
+  private _predictor: Predictor
+
+  constructor(
+    bot: BotDefinition,
+    _nluClient: NLUClient,
+    _defRepo: DefinitionsRepository,
+    _models: ModelEntryService,
+    _trainings: TrainingEntryService,
+    logger: sdk.Logger,
+    webSocket: (ts: TrainingSession) => void
```

Comment:
**nitpick:** I'm not loving webSocket naming as this is much more a `trainingSessionCallback` than an actual webSocket connection

Reply:
I agree, but at the same time I don't want to loose my approval and wait 30 minutes of Github Checks to merge this branch. I also just finished another round of QA.

So ill merge like this